Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -22,460 +22,653 @@
 #define FACECONVERTER_H
 
 #include "CarveHeaders.h"
-
 #include <memory>
-
 #include "ConverterBase.h"
-
 #include "CurveConverter.h"
 #include "PlacementConverter.h"
 #include "GeomUtils.h"
 #include "GeometryInputData.h"
 #include "PlacementConverter.h"
-
 #include "BlueFramework/Core/Diagnostics/log.h"
 
-
 namespace OpenInfraPlatform {
 	namespace Core {
 		namespace IfcGeometryConverter {
 			class ItemData;
 
+			/*! \brief Converter functionality for \c IfcSurfaces's subtypes.
+			*
+			* This class includes converter functions for
+			* \c IfcBoundedSurface and its subtypes,
+			* \c IfcElementarySurface and its subtypes and
+			* \c IfcSweptSurface.
+			*
+			* \param IfcEntityTypesT The IFC version templates
+			*/
+
 			template <
 				class IfcEntityTypesT
 			>
-			class FaceConverterT : public ConverterBaseT<IfcEntityTypesT>
+				class FaceConverterT : public ConverterBaseT<IfcEntityTypesT>
 			{
 			public:
+
+				//! Constructor
 				FaceConverterT(std::shared_ptr<GeometrySettings> geomSettings,
 					std::shared_ptr<UnitConverter<IfcEntityTypesT>> unitConverter,
 					std::shared_ptr<PlacementConverterT<IfcEntityTypesT>> pc,
 					std::shared_ptr<CurveConverterT<IfcEntityTypesT>> cc)
-					: 
+					:
 					ConverterBaseT<IfcEntityTypesT>(geomSettings, unitConverter),
 					placementConverter(pc),
 					curveConverter(cc)
 				{
 				}
 
+				//! Virtual destructor
 				virtual ~FaceConverterT()
 				{
 				}
 
-				void convertIfcSurface(const std::shared_ptr<typename IfcEntityTypesT::IfcSurface>& surface,
+				/*! \brief Converts \c IfcSurface to a according to subtype.
+					\param		surface	\c IfcSurface entity to be interpreted.
+					\param		pos
+					\param		polylineData
+					\note		The \c IfcSurface subtypes are \c IfcBoundedSurface,  \c IfcElementarySurface and  \c IfcSweptSurface.
+				*/
+
+				void convertIfcSurface(const EXPRESSReference<typename IfcEntityTypesT::IfcSurface>& surface,
 					const carve::math::Matrix& pos,
-					std::shared_ptr<carve::input::PolylineSetData>& polyline_data)

```

Comment:
Please change return types to the carve out parameters, in this case pos. Also, pos can't be const.

Reply:
`pos` should of course be const. It is the position of the geometry, as specified within `IfcObjectPlacement` of the product having this representation. Why can't it be const?

Reply:
I agree with the return.

Reply:
Additionally, all functions should be `const throw(...)`

Reply:
Resolved: 
const throws https://github.com/tumcms/Open-Infra-Platform/pull/151/commits/3055ee85eafc0b3ef5995d57b356a6ca510274a8 and
returns https://github.com/tumcms/Open-Infra-Platform/pull/151/commits/11078a11f06736b89bc3c4fa0e901d82d5cf67ef and returns https://github.com/tumcms/Open-Infra-Platform/pull/151/commits/1ee08611e218ac8b03e6fabd0e214cad06a096b1